packages/payment-detection/test/erc20/fee-proxy-contract.test.ts

```diff
@@ -185,61 +188,88 @@ describe('api/erc20/fee-proxy-contract', () => {
       version: '0.2',
     };
 
-    const mockExtractBalanceAndEvents = () => {
-      return Promise.resolve({
-        balance: '1000',
-        events: [
-          // Wrong fee address
+    const mockExtractTransferEvents = (eventName: any) => {
+      if (eventName === 'refund') {
+        return Promise.resolve([
+          // wrong fee address, ignore
           {
-            amount: '0',
-            name: PaymentTypes.EVENTS_NAMES.PAYMENT,
+            amount: '10',
+            name: PaymentTypes.EVENTS_NAMES.REFUND,
             parameters: {
               block: 1,
               feeAddress: 'fee address',
-              feeAmount: '5',
-              to: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
-              txHash: '0xABC',
+              feeAmount: '0',
+              to: '0xrefundAddress',
             },
-            timestamp: 10,
           },
-          // Correct fee address and a fee value
+          // valid refund
           {
-            amount: '500',
-            name: PaymentTypes.EVENTS_NAMES.PAYMENT,
+            amount: '10',
+            name: PaymentTypes.EVENTS_NAMES.REFUND,
             parameters: {
               block: 1,
               feeAddress: '0xC5fdf4076b8F3A5357c5E395ab970B5B54098Fef',
-              feeAmount: '5',
-              to: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
-              txHash: '0xABCD',
-            },
-            timestamp: 11,
-          },
-          // No fee
-          {
-            amount: '500',
-            name: PaymentTypes.EVENTS_NAMES.PAYMENT,
-            parameters: {
-              block: 1,
-              feeAddress: '',
               feeAmount: '0',
-              to: '0xf17f52151EbEF6C7334FAD080c5704D77216b732',
-              txHash: '0xABCDE',
+              to: '0xrefundAddress',
             },
-            timestamp: 12,
           },
-        ],
-      });
+        ]);
+      }
+      return Promise.resolve([
+        // Wrong fee address
+        {
+          amount: '0',
```

Comment:
```suggestion
          amount: '100',
```
Should not change the result if the fee address is wrong